### ISSUE TITLE
ValueError: Circular reference detected

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2260]
_Imported from trac ticket 2260.
Created by **kbower**
Opened in trac: Apr 2, 2013, 7:47:52 AM
Last modified in trac: Apr 2, 2013, 1:45:31 PM_
***

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 2, 2013, 7:48:11 AM:_

at El Dorado

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 2, 2013, 8:08:13 AM:_

After this morning's upgrade, I noticed I couldn't open many of the tickets with the automatic adjustments: I get a "ValueError: Circular reference detected"

I'm not sure if this is something the upgrade broke or not.  You mentioned you'd start looking at them yesterday... did you change anything on those records in GERS that could possibly have confused Loft?

There is not much to look at on TRAIN, where I'd ideally recreate the issue.

(Maybe it's time to refresh TRAIN from LIVE?  Then I could figure out what is going on there.)

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 2, 2013, 10:45:01 AM:_

`ValueError: Circular reference detected: Sale: orderid=[0305309JB87] customerid=[PENDI90326] orderdate=[2013-03-05]`

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 2, 2013, 11:21:36 AM:_

It's caused by `tags` backref: `order`
```
[Tue Apr 02 11:13:47 2013] [error] json property: tags
[Tue Apr 02 11:13:47 2013] [error] json property: order
[Tue Apr 02 11:13:47 2013] [error] json crash: <Sale: orderid=[0305309JB87] customerid=[PENDI90326] orderdate=[2013-03-05]>
```

**Comment by kentbower** [comment]
_**kbower** trac comment on Apr 2, 2013, 1:45:31 PM:_

(Changeset [[3989]](../commit/50e4d21fa2f0847b9e46d3a46140489b46244423))
* fixes #2260, ValueError: Circular reference detected for closed orders with tags
 * since tags are cloned when closed orders reconstructed, we need to skip concurrent checks (these are not very useful anyway since the record is just a primary key)
 * fix ordercomment fk to be initially deferred for closed order adjustments comments

**Comment by kentbower** [comment]
_**kbower** trac status change on Apr 2, 2013, 1:45:31 PM:_ **closed**
